### PR TITLE
LPS-187444 | Poshi Automation for Bug LPS-178900  | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# Poshi Automation for Bug LPS-178900

Hi! :)
This pr is corresponding to this ticket: [LPS-178900](https://liferay.atlassian.net/browse/LPS-178900) → [LPS-187444](https://liferay.atlassian.net/browse/LPS-187444)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase [hunk 1]
```diff
@@ -1010,4 +1010,55 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-187444. Deleted experience resets widget configuration on default experience"
+	@priority = 4
+	test ViewWebcontentPersistsAfterDeleteExperience {
+		task ("Given a content page with an asset publisher widget") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase:1016]: Add property `property asset.publisher.selection.style = "asset-list";`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase [hunk 2]
```diff
@@ -1010,4 +1010,55 @@ definition {
...
+			JSONWebcontent.addWebContent(
+				content = "Web Content Content",
+				groupName = ${randomSiteName},
+				title = "Web Content Title 1");
+
+			ContentPagesNavigator.openEditContentPage(
+				pageName = "Test Page Name",
+				siteName = ${randomSiteName});
+
+			AssertElementPresent(locator1 = "PageEditor#EXPERIENCE_DEFAULT_BUTTON_NAME");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase:1026]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase [hunk 3]
```diff
@@ -1010,4 +1010,55 @@ definition {
...
+
+			PageEditor.addWidget(portletName = "Asset Publisher");
+
+			AssetPublisherPortlet.configureAssetTypePG(
+				assetSubtype = "Any",
+				selectedAsset = "Web Content Article");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase:1032]: AssetPublisherPortlet.configureDynamicAssetSelectionPG

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase [hunk 4]
```diff
@@ -1010,4 +1010,55 @@ definition {
...
+
+			IFrame.closeFrame();
+
+			PageEditor.publish();
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase:1037]: Add step to navigate to content page at view mode

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase (hunk 1 above)
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase:1016]: Given a page designer has an experienced content page with an Asset Publisher widget

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase [hunk 5]
```diff
@@ -1010,4 +1010,55 @@ definition {
...
+			AssetPublisherPortlet.viewAssetPG(
+				assetContent = "Web Content Content",
+				assetTitle = "Web Content Title 1");
+		}
+
+		task ("When create and delete an experience") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase:1043]: When the page designer deletes the experience

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase [hunk 6]
```diff
@@ -1010,4 +1010,55 @@ definition {
...
+			PageEditor.addExperience(
+				experienceName = "New Experience",
+				segmentName = "Anyone");
+
+			PageEditor.deleteExperience(experienceName = "New Experience");
+
+			Refresh();
+		}
+
+		task ("Then the content of the Asset publisher widget must persist on the page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase:1057]: Then the page designer still can see the web content displayed in Asset Publisher

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase [hunk 7]
```diff
@@ -1010,4 +1010,55 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-187444. Deleted experience resets widget configuration on default experience"
+	@priority = 4
+	test ViewWebcontentPersistsAfterDeleteExperience {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPagesWithExperiencePersonalization.testcase:1015]: ViewDisplayedWebContentPersistedInAssetPublisherAfterDeleteExperience

**Review by Tim-Cao — COMMENTED** [review]
Hi @rafaellajordao, would you mind refactoring according to comments? Thanks

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
@rafaellajordao  See changes https://github.com/liferay-echo/liferay-portal/pull/12625

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**